notebooks/official/tensorboard/tensorboard_custom_training_with_custom_container.ipynb

```diff
@@ -0,0 +1,1024 @@
+{
```

Comment:
<p>The assumption of <code>SA_EMAIL = "{}@{}.iam.gserviceaccount.com".format(SERVICE_ACCOUNT, PROJECT_ID)</code> won't work for many of the cases so we expect that  SERVICE_ACCOUNT is given as a service account email. If a service account is not specified, we can use the default service account for compute so no need to create a new account.</p><p>Please see example below:</p><pre class="ql-syntax" spellcheck="false">SERVICE_ACCOUNT = "[your-service-account]"  # @param {type:"string"}
</pre><p>In&nbsp;[&nbsp;]:</p><pre class="ql-syntax" spellcheck="false">if (
    SERVICE_ACCOUNT == ""
    or SERVICE_ACCOUNT is None
    or SERVICE_ACCOUNT == "[your-service-account]"
):
    # Get your service account from gcloud
    if not IS_COLAB:
        shell_output = ! gcloud auth list 2&gt;/dev/null
        SERVICE_ACCOUNT = shell_output[2].replace("*", "").strip()

    else:  # IS_COLAB:
        shell_output = ! gcloud projects describe  $PROJECT_ID
        project_number = shell_output[-1].split(":")[1].strip().replace("'", "")
        SERVICE_ACCOUNT = f"{project_number}-compute@developer.gserviceaccount.com"

    print("Service Account:", SERVICE_ACCOUNT)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/611/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Reply:
<p>resolved in the same way as pre-built container</p>